scipy/interpolate/_ndgriddata.py

```diff
@@ -92,12 +92,17 @@ class NearestNDInterpolator(NDInterpolatorBase):
     def __init__(self, x, y, rescale=False, tree_options=None):
         NDInterpolatorBase.__init__(self, x, y, rescale=rescale,
                                     need_contiguous=False,
-                                    need_values=False)
+                                    need_values=True)
```

Comment:
Can we remove this too? This is not needed by vanilla scipy interpolators; a downstream subclass either knows it needs it or can add this argument ISTM.

Reply:
I think it can be removed, but won't it break existing usages of the current subclasses which use this attribute?
Would you like me to deprecate it somehow?
If so, can you direct me at an example of how deprecation works in scipy?

Reply:
Ah, yes, the parameter was there, no need to deprecate it!